File: src/main/java/com/github/xumeng/java/problems/easy/problem0231/Solution0231S1.java
package com.github.xumeng.java.problems.easy.problem0231;

public class Solution0231S1 {

    public boolean isPowerOfTwo(int n) {
        if (n <= 0) {
            return false;
        }
        for (int i = 0; i < 32; i++) {
            double power = Math.pow(2, i);
            if (power == n) {
                return true;
            }
            if (power > n) {
                return false;
            }
        }
        return false;
    }

}

File: src/main/kotlin/com/github/xumeng/kotlin/problems/easy/problem0231/Solution0231S1.kt
package com.github.xumeng.kotlin.problems.easy.problem0231

import kotlin.math.pow

class Solution0231S1 {

    fun isPowerOfTwo(n: Int): Boolean {
        if (n <= 0) {
            return false
        }
        for (i in 0..<32) {
            val power = 2.0.pow(i)
            if (power == n.toDouble()) {
                return true
            }
            if (power > n) {
                return false
            }
        }
        return false
    }

}